acme/dns_challenge.go

```diff
@@ -262,9 +262,15 @@ func FindZoneByFqdn(fqdn string, nameservers []string) (string, error) {
 		if in.Rcode == dns.RcodeSuccess {
 			for _, ans := range in.Answer {
 				if soa, ok := ans.(*dns.SOA); ok {
-					zone := soa.Hdr.Name
-					fqdnToZone[fqdn] = zone
-					return zone, nil
+					if zone, err := publicsuffix.EffectiveTLDPlusOne(UnFqdn(fqdn)); err == nil {
```

Comment:
Wouldn't that lead to most lookups using `FindZoneByFqdn()` to use this branch instead of the original one - effectively bypassing the lookup?